templates/inspec/singular_resource.erb

```diff
@@ -35,6 +35,22 @@ class <%= object.name -%> < Inspec.resource(1)
     '<%= url(object) %>'
   end
 
+<% if object.self_link_query.nil? -%>
+  def initialize(params)
+    @fetched = fetch_resource(params)
+    parse unless @fetched.nil?
+  end
+<% else # object.self_link_query.nil? -%>
+   # TODO(slevenick) for other products
+   def initialize(params)
```

Comment:
Note: another initialize method will be needed here to support other products (SQL in particular) but I don't want to add the code here before it will be used to generate anything.